crates/telio-utils/src/git.rs

```diff
@@ -1,14 +1,23 @@
+use std::ffi::CStr;
+
 /// Get the commit sha that was used when building.
 /// If not present defaults to "dev".
 pub fn commit_sha() -> &'static str {
     option_env!("LIBTELIO_COMMIT_SHA").unwrap_or("dev")
 }
 
-/// Get the version tag that was used when building.
-/// If not present defaults to "dev".
+/// Get the version placeholder (half of the maximum length of git tag)
+/// Will be replaced during build promotions
 pub fn version_tag() -> &'static str {
-    option_env!("LIBTELIO_COMMIT_TAG")
-        .map(|tag| tag.trim())
-        .and_then(|tag| if tag.is_empty() { None } else { Some(tag) })
-        .unwrap_or("dev")
+    // Attempt to create a CStr from bytes with null termination
+    let cstr = match CStr::from_bytes_with_nul(b"VERSION_PLACEHOLDER@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@\0") {
+        Ok(cstr) => cstr,
+        Err(_) => {
+            return "";
+        }
+    };
+    match cstr.to_str() {
+        Ok(str_ref) => str_ref,
+        Err(_) => "",
```

Comment:
nit: I think it would be better to return two different strings in both error cases so that if it ever happens, we know immediately which one of the errors happened.